File: a2/q1/BankCli.py
# Created by:	Colin Dyson
# Student #:	7683407

import sys
import socket

sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
host = "cormorant.cs.umanitoba.ca"
port = 13059
sock.settimeout(10)

print("Client starting")

exit = False

while exit == False:
	try:
		term_in = input()
	except Exception:
		print("Failed to read from terminal.")
		sys.exit()
	if term_in != "E":
		try:
			sock.connect((host, port))
		except socket.error as e:
			print("Failed to create socket.", e)
			sys.exit()
		try:
			sock.send(term_in.encode())
			response = sock.recv(1024)
			print("[Server]:", response)
		except Exception:
			print("Communication with the server has failed.")
			sys.exit()
	else:
		exit = True

try:
	sock.close()
except Exception:
	print("Client couldn't close socket")
	sys.exit()

print("Client finished")

File: a2/q1/BankCli.pl
# Created by:	Colin Dyson
# Student #:	7683407
use IO::Handle;
use strict;
use warnings;
use Socket;

my $sock;
my $host = "cormorant.cs.umanitoba.ca";
my $port = 13059;

my $iaddr = inet_aton($host) || die "no host: $host";
my $paddr = sockaddr_in($port, $iaddr);
my $proto = getprotobyname("tcp");

my $exit = 0;
my $in;
my $response;

print "Client Starting\n";

while ($exit != 1) {
	print ">";
	$in = <STDIN>;
	chop $in;

	if ($in ne "E") {
		socket($sock, PF_INET, SOCK_STREAM, $proto) || die  "Failed to create socket";
		$sock->autoflush;
		connect($sock, $paddr) || die "Failed to connect to server";
		print $sock "$in\n" || die "Communication with the server has failed.";
		$response = <$sock>;
		print "[Server]: $response\n";
		close $sock || die "Failed to close socket";
	} else {
		$exit = 1;
	}
}

print "Client Finished.\n";
